### PR TITLE
Configuring Data doc

### LINKED ISSUE #44: api-*.json and api.json documentation

### PR DESCRIPTION
**Pull request by pragmaticivan**
closes #44

### CODE REVIEW

**Comment by pragmaticivan** [comment]
I will need some help here cc @eduardolundgren @laissandrade There's no doc about data in general, so I just added what I found.

**Comment by laissandrade** [comment]
Hey, I can add more information to data docs later and send a PR to you, so you can format it to follow our docs pattern. How does that sound? I can create an issue for this if you agree.

**Comment by pragmaticivan** [comment]
Sounds good :D

File: data/js/configuring-data.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+## Understanding configuration files
+
+By default WeDeploy Data service is going to use all the JSON files starting with `api-*` and also the file `api.json`.
+
+These files are used to help you manage features such as path validation, authentication, and params validation.
+
+The api JSON files are located at the same path of the `./container.json` and are used following the ordering filesystem.
+
+By the default, if you don't add any api.json file all the collections are going to be exposed.
```
Comment (laissandrade) [inline data/js/configuring-data.md]: I'm not sure about this. We don't use boilerplate when we build a container, do we? I though it was used only when we install a data container via dashboard, without git.

File: data/js/configuring-data.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+## JSON attributes
+
+After understanding how the api configuration files work, it's time to learn what are the supported attributes:
+
+<table class="table">
+  <tr>
+    <th>Field</th> <th>Description</th>
+  </tr>
+  <tr>
+    <td>path</td> <td>The path that represents the collection used to handle the request data.</td>
```
Comment (laissandrade) [inline data/js/configuring-data.md:28]: The path if the full resource path that we want to access/modify with this request. We may describe here the pattern: `/collectionName/id/field/nestedField/...`

File: data/js/configuring-data.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  </tr>
+  <tr>
+    <td>data</td> <td>Tells the service if the request to a collection should be stored or not.</td>
```
Comment (laissandrade) [inline data/js/configuring-data.md:31]: Maybe inform that this tells if this endpoint hits the datastore or not? I'm not sure this makes clear that the matching will end here and request will end when it hits this endpoint.

File: data/js/configuring-data.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  </tr>
+  <tr>
+    <td>description</td> <td>Used to describe the behavior of an endpoint.</td>
+  </tr>
+  <tr>
+    <td>auth</td> <td>Used to define authentication rules for the collection.</td>
```
Comment (laissandrade) [inline data/js/configuring-data.md]: "...authentication rules for requests that matches this endpoint's path"? Its not only collection-wise.

File: data/js/configuring-data.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  </tr>
+  <tr>
+    <td>method</td> <td>HTTP method allowed for the request.</td>
+  </tr>
+  <tr>
+    <td>parameters</td> <td>Params and validation rules for the collection.</td>
```
Comment (laissandrade) [inline data/js/configuring-data.md]: "Parameters validation rules..."?

File: data/js/configuring-data.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  </tr>
+</table>
+
+**path**
+A path represents the collection used to store your project data.
```
Comment (laissandrade) [inline data/js/configuring-data.md]: The path is full resource path, not only collections.

File: data/js/configuring-data.md (hunk 1 above)
Comment (pragmaticivan) [reply data/js/configuring-data.md]: Don't we need a folder with a container.json to have data running? And is the `api.json` inside the same folder?

File: data/js/configuring-data.md (hunk 6 above)
Comment (pragmaticivan) [reply data/js/configuring-data.md]: Are we not mixing engine concepts with data concepts? I'm just talking about data so far. User should not know about resource path. We're not talking about engine in this case.

File: data/js/configuring-data.md (hunk 1 above)
Comment (laissandrade) [reply data/js/configuring-data.md]: Yes, if we create this container with github on Deployment. But i believe in this case we don't have the `By the default` behavior, as it comes from boilerplate-data that is overridden in built containers with user's files (or lack thereof).

File: data/js/configuring-data.md (hunk 6 above)
Comment (laissandrade) [reply data/js/configuring-data.md]: Yes, but this is for data. If you have the fruits collection and you do `GET /fruits/1/calories`, you get the integer stored in this field in your response body. You can add/get/update/delete any resource stored in data like this:

```
DELETE /fruits/1/name/scientificName
PUT /fruits/1/price
...
```

Comment (pragmaticivan) [reply data/js/configuring-data.md]: Cool, understood. I was missing that about resources.

**Comment by laissandrade** [comment]
I think its better to merge this for now and I can add information about operators and all those suggestions I added here on top of this. wdyt @zenorocha @pragmaticivan?

**Pull request merged**